### PR TITLE
Server test

### PR DESCRIPTION
**Pull request by jackcarlisle**
## ready for review
* adds tests for the server  
#22

### CODE REVIEW

File: examples/direct-upload/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    if (err) {
+      return next(err)
+    }
+
+    server.start(function (err) {
+      if (err) console.log('error starting server: ', err)
+      return next(err, server)
+    })
+  })
+  module.exports = server
```
Comment (simon-p-r) [inline examples/direct-upload/lib/index.js]: Remove this line

File: examples/direct-upload/lib/start.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+var Server = require('./index.js')
+var Hoek = require('hoek')
+
+Server.init(process.env.PORT || 8000, function (err, server) {
```
Comment (simon-p-r) [inline examples/direct-upload/lib/start.js]: Add this function block to your test script to access server.inject

File: examples/direct-upload/server.js [hunk 3]
```diff
@@ -21,43 +21,47 @@ server.register(require('inert'),
...
 
 server.start((err) => {
   if (err) {
     throw err
   }
-  console.log(`Server running at: ${server.info.uri}`)
+  console.log(`✅ Server running at: ${server.info.uri}`)
 })
+
+module.exports = server
```
Comment (simon-p-r) [inline examples/direct-upload/server.js]: Remove this line

File: examples/direct-upload/test/server.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+var test = require('tape')
+var server = require('../lib/index.js')
+
+test('simple server running', function (t) {
+  var options = {
+    method: 'GET',
+    url: '/'
+  }
+
+  server.inject(options, function (response) {
```
Comment (simon-p-r) [inline examples/direct-upload/test/server.test.js]: Add server.init function here and inside callback access server.inject for testing, see this example

https://github.com/davethomas11/stackoverlfow_Q_39432656/blob/master/test/index.js#L24

Comment (jackcarlisle) [inline examples/direct-upload/test/server.test.js]: @simon-p-r thank you! I'll try it out and then let you know

**Comment by simon-p-r** [comment]
Just saw your issue on hapi, so made some comments to get your tests working.

**Comment by jackcarlisle** [comment]
@simon-p-r I've changed my server by moving it all into the same file. Can you see where I'm going wrong with that one? https://github.com/hapijs/discuss/issues/361 I don't have the `server.init` function anymore

**Comment by simon-p-r** [comment]
Yeah saw that, I would not ever use ```module.exports``` to export server as you have no control on the state of object.  Follow the structure at https://github.com/davethomas11/stackoverlfow_Q_39432656 for an idea how to basically structure a hapi based project.  It allows you to have a simple interface you can test sanely.

**Comment by jackcarlisle** [comment]
@simon-p-r ok I'll check it out, thanks for all your help!

File: examples/direct-upload/README.md [hunk 5]
```diff
@@ -421,8 +423,8 @@ server.register(require('inert'),
     path: '/{filename*}',
     handler: {
       directory: {
-        path: __dirname + '/public',
-        listing: false,
+        path: 'public',
```
Comment (nelsonic) [inline examples/direct-upload/README.md]: does this _work_ without path resolution? (_guessing it does... didn't know..._)

**Comment by nelsonic** [comment]
Looks good @jackcarlisle 👍  (_thanks_!)

**Pull request merged**